src/generators.rs

```diff
@@ -76,14 +75,13 @@ pub fn cycle_graph(
     })
 }
 
-/// Generate a directed cycle graph
+/// Generate a directed cycle graph.
 ///
 /// :param int num_nodes: The number of nodes to generate the graph with. Node
 ///     weights will be None if this is specified. If both ``num_nodes`` and
-///     ``weights`` are set this will be ignored and ``weights`` will be used.
-/// :param list weights: A list of node weights, the first element in the list
-///     will be the center node of the cycle graph. If both ``num_nodes`` and
-///     ``weights`` are set this will be ignored and ``weights`` will be used.
+///     ``weights`` are set, this will be ignored and ``weights`` will be used.
+/// :param list[S] weights: A list of node weights. If both ``num_nodes`` and
```

Comment:
```suggestion
/// :param Sequence[Any] weights: A sequence of node weights, typically a list. If both ``num_nodes`` and
```
Source: https://github.com/Qiskit/rustworkx/blob/c21e3d365318015d9aced9c4688b429421f3a7e3/rustworkx/generators/__init__.pyi#L19